types/AsResourceTest.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+use function PHPStan\Testing\assertType;
+
+$resource = fopen('php://memory', 'r');
+$variable = random_int(0, 1) !== 0 ? 'string' : $resource;
+assertType('resource', $variable = type($variable)->asResource());
```

Comment:
```suggestion
assertType('resource', type($variable)->asResource());
```

Reply:
I used the typed variable as the resource can also be `false`